File: app/src/main/java/com/test/mercaridemo/shoppinglist/di/ShoppingListFragmentBuilderModule.java
package com.test.shoppingapp.shoppinglist.di;

import com.test.shoppingapp.shoppinglist.fragment.ItemFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ShoppingListFragmentBuilderModule {

    @ContributesAndroidInjector
    abstract ItemFragment bindItemFragment();
}

File: app/src/main/java/com/test/shoppingapp/shoppinglist/di/ShoppingListFragmentBuilderModule.kt
package com.test.shoppingapp.shoppinglist.di

import com.test.shoppingapp.shoppinglist.fragment.ShoppingListFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ShoppingListFragmentBuilderModule {

    @ContributesAndroidInjector
    internal abstract fun bindItemFragment(): ShoppingListFragment
}
